modules/library/src/main/scala/zio/elasticsearch/query/Queries.scala

```diff
@@ -109,6 +110,101 @@ private[elasticsearch] final case class Exists[S](field: String) extends ExistsQ
+   * circle are considered to be matches. The distance can be specified in various units. See
+   * [[zio.elasticsearch.query.DistanceUnit]].
+   *
+   * @param value
+   *   the [[Double]] value for distance parameter
+   * @param unit
+   *   the [[zio.elasticsearch.query.DistanceUnit]] in which we want to represent the distance
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.GeoDistanceQuery]] with the score value set.
+   */
+  def distance(value: Double, unit: DistanceUnit): GeoDistanceQuery[S]
+
+  /**
+   * sets the `distanceType` parameter for the [[zio.elasticsearch.query.GeoDistanceQuery]]
+   *
+   * Defines how to compute the distance. Can either be [[zio.elasticsearch.query.DistanceType.Arc]] (default), or
+   * [[zio.elasticsearch.query.DistanceType.Plane]] (faster, but inaccurate on long distances and close to the poles).
+   *
+   * @param value
+   *   the [[zio.elasticsearch.query.DistanceType]] value to represent distance type
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.GeoDistanceQuery]] with the score value set.
+   */
+  def distanceType(value: DistanceType): GeoDistanceQuery[S]
+
+  /**
+   * sets the `queryName` parameter for the [[zio.elasticsearch.query.GeoDistanceQuery]]
+   *
+   * Represents the optional name field to identify the query
+   *
+   * @param value
+   *   the [[String]] value to represent the name field
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.GeoDistanceQuery]] with the score value set.
+   */
+  def name(value: String): GeoDistanceQuery[S]
+
+  /**
+   * sets the `validationMethod` parameter for the [[zio.elasticsearch.query.GeoDistanceQuery]]
+   *
+   * Set to [[zio.elasticsearch.query.ValidationMethod.IgnoreMalformed]] to accept geo points with invalid latitude or
+   * longitude, set to [[zio.elasticsearch.query.ValidationMethod.Coerce]] to additionally try and infer correct
+   * coordinates (default is [[zio.elasticsearch.query.ValidationMethod.Strict]]).
+   *
+   * @param value
+   *   the [[zio.elasticsearch.query.ValidationMethod]] value to represent the validationMethod field
+   * @return
+   *   a new instance of the [[zio.elasticsearch.query.GeoDistanceQuery]] with the score value set.
+   */
+  def validationMethod(value: ValidationMethod): GeoDistanceQuery[S]
+}
+
+private[elasticsearch] final case class GeoDistance[S](
+  field: String,
+  point: Either[(Double, Double), String],
+  distance: Option[Distance] = None,
+  distanceType: Option[DistanceType] = None,
+  queryName: Option[String] = None,
+  validationMethod: Option[ValidationMethod] = None
+) extends GeoDistanceQuery[S] { self =>
+
+  def distance(value: Double, unit: DistanceUnit): GeoDistanceQuery[S] =
+    self.copy(distance = Some(Distance(value, unit)))
+
+  def distanceType(value: DistanceType): GeoDistanceQuery[S] = self.copy(distanceType = Some(value))
+
+  private def getPointJson: (String, Json) = point match {
```

Comment:
I would move this to `paramsToJson` if it's used only once. Also, no need for a method.